server/service/src/vvm/vvm_status_log/insert/validate.rs

```diff
@@ -0,0 +1,45 @@
+use repository::{
+    vvm_status::{
+        vvm_status_log_row::VVMStatusLogRowRepository, vvm_status_row::VVMStatusRowRepository,
+    },
+    RepositoryError, StorageConnection,
+};
+
+use crate::common_stock::check_stock_line_does_not_exist;
+
+use super::{InsertVVMStatusLogError, InsertVVMStatusLogInput};
+
+pub fn validate(
+    input: &InsertVVMStatusLogInput,
+    connection: &StorageConnection,
+) -> Result<(), InsertVVMStatusLogError> {
+    if check_vvm_status_log_exists(&input.id, connection)? {
+        return Err(InsertVVMStatusLogError::VVMStatusLogAlreadyExists);
+    }
+
+    if !check_vvm_status_exists(&input.status_id, connection)? {
+        return Err(InsertVVMStatusLogError::VVMStatusDoesNotExist);
+    }
+
+    if check_stock_line_does_not_exist(&input.stock_line_id, connection)? {
+        return Err(InsertVVMStatusLogError::StockLineDoesNotExist);
+    }
+
+    Ok(())
+}
+
+pub fn check_vvm_status_log_exists(
```

Comment:
This and `check_vvm_status_exists` could be moved to common to use when upserting invoice_lines, but can worry about that in the issue doing it (: